encoding/template/run_length_decode.go

```diff
@@ -0,0 +1,72 @@
+package template
+
+import (
+	"encoding/binary"
+
+	"github.com/xichen2020/eventdb/x/io"
+)
+
+// WriteValueFn reads a GenericValue from an `io.Reader`.
+type WriteValueFn func(reader io.Reader) (GenericValue, error)
+
+// runLengthDecodeValue run length decodes a stream of GenericValues.
+func runLengthDecodeValue(
+	reader io.Reader,
+	writeValue WriteValueFn,
```

Comment:
Shouldn't this be `readValue ReadValueFn`? 

If so, you should move `ReadValueFn` to `run_length_encode.go` and `WriteValueFn` to `run_length_decode.go`.

Reply:
👍